sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -25,48 +25,57 @@
 public final class ActivityFramesTracker {
 
   private @Nullable FrameMetricsAggregator frameMetricsAggregator = null;
-  private boolean androidXAvailable = true;
+  private final SentryAndroidOptions options;
 
   private final @NotNull Map<SentryId, Map<String, @NotNull MeasurementValue>>
       activityMeasurements = new ConcurrentHashMap<>();
   private final @NotNull Map<Activity, FrameCounts> frameCountAtStartSnapshots =
       new WeakHashMap<>();
 
-  private final @Nullable ILogger logger;
   private final @NotNull MainLooperHandler handler;
 
   public ActivityFramesTracker(
       final @NotNull LoadClass loadClass,
-      final @Nullable ILogger logger,
+      final @NotNull SentryAndroidOptions options,
       final @NotNull MainLooperHandler handler) {
-    androidXAvailable =
-        loadClass.isClassAvailable("androidx.core.app.FrameMetricsAggregator", logger);
+
+    final boolean androidXAvailable =
+        loadClass.isClassAvailable("androidx.core.app.FrameMetricsAggregator", options.getLogger());
+
     if (androidXAvailable) {
       frameMetricsAggregator = new FrameMetricsAggregator();
     }
-    this.logger = logger;
+    this.options = options;
     this.handler = handler;
   }
 
-  public ActivityFramesTracker(final @NotNull LoadClass loadClass, final @Nullable ILogger logger) {
-    this(loadClass, logger, new MainLooperHandler());
+  public ActivityFramesTracker(
+      final @NotNull LoadClass loadClass, final @NotNull SentryAndroidOptions options) {
+    this(loadClass, options, new MainLooperHandler());
   }
 
+  /**
+   * @deprecated please use {{@link ActivityFramesTracker#ActivityFramesTracker(LoadClass,
+   *     SentryAndroidOptions)}} instead.
+   */
   public ActivityFramesTracker(final @NotNull LoadClass loadClass) {
-    this(loadClass, null);
+    this(loadClass, new SentryAndroidOptions());
```

Comment:
Creating a new options here won't help, the feature will be always enabled and can't be disabled since `options` is private, maybe better to just remove this ctor in this case and fix on Flutter/RN if necessary.

Reply:
> the feature will be always enabled 
 
exactly, that matches the existing behavior thus I thought it makes sense. But in hindsight it's actually more confusing than helpful, let me remove this again 👍